d2graph/d2graph.go

```diff
@@ -1160,15 +1148,36 @@ func (g *Graph) SetDimensions(mtexts []*d2target.MText, ruler *textmeasure.Ruler
 			sideLength := math.Max(obj.Width+paddingX, obj.Height+paddingY)
 			obj.Width = sideLength
 			obj.Height = sideLength
-
-		default:
+		} else {
 			if desiredWidth == 0 {
 				obj.Width += float64(paddingX)
 			}
 			if desiredHeight == 0 {
 				obj.Height += float64(paddingY)
 			}
 		}
+
+		// give shapes with icons extra padding
+		if obj.Attributes.Icon != nil {
+			paddingX += 20
+			paddingY += 20
```

Comment:
this also kinda depends on font size (cuz of line height). do you have access to shapes label height here? if so we should just use that instead of 20